pkg/handlers/handlers.go

```diff
@@ -56,3 +63,151 @@ func HandleFile(ctx context.Context, file io.Reader, chunkSkel *sources.Chunk, c
+		return nil, err
+	}
+	defer os.Remove(tempEnv.tempFileName)
+	defer os.RemoveAll(tempEnv.extractPath)
+
+	cmd := exec.Command("ar", "x", tempEnv.tempFile.Name())
+	cmd.Dir = tempEnv.extractPath
+	if err := executeCommand(cmd); err != nil {
+		return nil, err
+	}
+
+	// List the content of the extraction directory.
+	extractedFiles, err := os.ReadDir(tempEnv.extractPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read extracted directory: %w", err)
+	}
+
+	// Determine the correct data archive name. (e.g., data.tar.gz, data.tar.xz)
+	var dataArchiveName string
+	for _, file := range extractedFiles {
+		if strings.HasPrefix(file.Name(), "data.tar.") {
+			dataArchiveName = file.Name() // Use the actual name if different
+			break
+		}
+	}
+
+	return openDataArchive(tempEnv.extractPath, dataArchiveName)
+}
+
+// extractRpmContent takes an .rpm file as an io.ReadCloser, extracts its contents
+// into a temporary directory, and returns a ReadCloser for the extracted data archive.
+// It handles the extraction process by using the 'rpm2cpio' and 'cpio' commands and manages temporary
+// files and directories for the operation.
+// The caller is responsible for closing the returned reader.
+func extractRpmContent(ctx logContext.Context, file io.ReadCloser) (io.ReadCloser, error) {
+	tempEnv, err := createTempEnv(file)
+	if err != nil {
+		return nil, err
+	}
+	defer os.Remove(tempEnv.tempFileName)
+	defer os.RemoveAll(tempEnv.extractPath)
+
+	// Use rpm2cpio to convert the RPM file to a cpio archive and then extract it using cpio command.
+	cmd := exec.Command("sh", "-c", "rpm2cpio "+tempEnv.tempFile.Name()+" | cpio -id")
+	cmd.Dir = tempEnv.extractPath
+	if err := executeCommand(cmd); err != nil {
+		return nil, err
+	}
+
+	// List the content of the extraction directory.
+	extractedFiles, err := os.ReadDir(tempEnv.extractPath)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read extracted directory: %w", err)
+	}
+
+	var dataArchiveName string
+	// Determine the correct data archive name.
+	for _, file := range extractedFiles {
+		if strings.HasSuffix(file.Name(), ".tar.gz") {
+			dataArchiveName = file.Name()
+			break
+		}
+	}
+
+	return openDataArchive(tempEnv.extractPath, dataArchiveName)
+}
+
+type tempEnv struct {
+	tempFile     *os.File
+	tempFileName string
+	extractPath  string
+}
+
+// createTempEnv creates a temporary file and a temporary directory for extracting archives.
+// The caller is responsible for removing these temporary resources
+// (both the file and directory) when they are no longer needed.
+func createTempEnv(file io.ReadCloser) (tempEnv, error) {
+	tempFile, err := os.CreateTemp("", "tmp")
+	if err != nil {
+		return tempEnv{}, fmt.Errorf("unable to create temporary file: %w", err)
+	}
+
+	extractPath, err := os.MkdirTemp("", "tmp_archive")
+	if err != nil {
+		return tempEnv{}, fmt.Errorf("unable to create temporary directory: %w", err)
+	}
+
+	if _, err = io.Copy(tempFile, file); err != nil {
```

Comment:
I don't think the file size has been checked, so this could result in copying a very large file.

Reply:
Is the idea here we just read to up `maxSize` or do we skip the file outright?

Reply:
I think reading up to `maxSize` is the way to go.